components/tidb_query_expr/src/impl_string.rs

```diff
@@ -679,9 +679,22 @@ pub fn substring_index(
 
 #[rpn_fn]
 #[inline]
-pub fn strcmp(left: BytesRef, right: BytesRef) -> Result<Option<i64>> {
+pub fn strcmp<C: Collator>(left: BytesRef, right: BytesRef) -> Result<Option<i64>> {
     use std::cmp::Ordering::*;
-    Ok(Some(match left.cmp(right) {
+    let left = match str::from_utf8(left) {
+        Ok(left) => left,
+        Err(err) => return Err(box_err!("invalid input value: {:?}", err)),
+    };
+    let right = match str::from_utf8(right) {
+        Ok(right) => right,
+        Err(err) => return Err(box_err!("invalid input value: {:?}", err)),
+    };
+    let offset = if C::IS_CASE_INSENSITIVE {
```

Comment:
`offset` is not a good name for result of `cmp`, please use `order` / `ord` instead.